ansible/roles/master/defaults/main.yml

```diff
@@ -1,7 +1,14 @@
-kube_master_insecure_port: 8080
+kube_api_insecure_port: 8080
+
+kube_master_api_port: 443
+
+kube_apiserver_interface: "{{ ansible_default_ipv4.interface }}"
+kube_apiserver_endpoint: "{% if groups['masters']|length > 1 %}{{ kube_apiserver_vip }}{% else %}{{ groups['masters'][0] }}{% endif %}"
```

Comment:
when removing kube_apiserver_endpoint, we shod not define kube_apiserver_vip here. kube_apiserver_vip should be exposes in group_vars/all since a user/operator is required to manually specify an available IP to assign. You can checkout out the openstack kolla project as they do something very similar with front-ending openstack api endpoints with a vip.